aptos-move/move-examples/mint_nft/step1/sources/nft_step1.move

```diff
@@ -0,0 +1,144 @@
+
+        // Create the nft collection.
+        token::create_collection(source_account, collection_name, description, collection_uri, maximum_supply, mutate_setting);
+
+        // Create a token data id to specify the token to be minted.
+        let token_data_id = token::create_tokendata(
+            source_account,
+            collection_name,
+            token_name,
+            string::utf8(b""),
+            0,
+            token_uri,
+            signer::address_of(source_account),
+            1,
+            0,
+            // we don't allow any mutation to the token
+            token::create_token_mutability_config(
+                &vector<bool>[ false, false, false, false, true ]
+            ),
+            vector::empty<String>(),
+            vector::empty<vector<u8>>(),
+            vector::empty<String>(),
+        );
+
+        // Store the token data id within the module, so we can refer to it later
+        // when we're minting the NFT and updating its property version.
+        move_to(source_account, ModuleData {
+            token_data_id,
+        });
+    }
+
+    /// Mint an NFT to the receiver. Note that here we ask two accounts to sign: the module owner and the receiver.
+    /// This is not ideal in production, because we don't want to manually sign each transaction. It is also
+    /// inpractical/weird in general, because we either need to implement delayed execution on our own, or have
+    /// two keys to sign at the same time.
+    /// In the step2 of this tutorial, we will introduce the concept of "resource account" - it is
+    /// an account controlled by smart contracts to automatically sign for transactions. Resource account is also known
+    /// as PDA or smart contract account in general blockchain terms.
+    public entry fun mint_nft(module_owner: &signer, receiver: &signer) acquires ModuleData {
+        // Assert that the module owner signer is the owner of this module.
+        assert!(signer::address_of(module_owner) == @mint_nft, error::permission_denied(ENOT_AUTHORIZED));
+
+        // Mint token to the receiver.
+        let module_data = borrow_global_mut<ModuleData>(@mint_nft);
+        let token_id = token::mint_token(module_owner, module_data.token_data_id, 1);
+        token::direct_transfer(module_owner, receiver, token_id, 1);
+
+        // Mutate the token properties to update the property version of this token.
+        // Note that here we are re-using the same token data id and only updating the property version.
+        // This is because we are simply printing edition of the same token, instead of creating
+        // tokens with unique names and token uris. The tokens created this way will have the same token data id,
+        // but different property versions.
+        let (creator_address, collection, name) = token::get_token_data_id_fields(&module_data.token_data_id);
+        token::mutate_token_properties(
```

Comment:
This seems more complex than necessary. Why do we need to include mutate_token_properties in this step? IMO this should be as simple as just minting one token. Mutating can perhaps be introduced in a later step.

Reply:
ah I see, makes sense

Reply:
ah so I was thinking that the step 1 will cover all the collection + token things, and then step 2 is resource account, step 3 is admin account / functions, and step 4 is getting production-ready (emitting custom events, signature verification, & unit tests). I feel like this is more related to step 1 and would be cleaner if we just cover it in the first step. like you mentioned earlier, I feel like people would overlook it if we add new code in a previously existing function.